README.md

```diff
@@ -1 +1,178 @@
-# php-wowemu-auth
\ No newline at end of file
+# php-wowemu-auth
+
+## Requirements
+
+* PHP 7.1+
+* Web server (ex. Apache or Nginx)
+* CMaNGOS instance
+
+## Installation
+
+You can install the libary via composer:
+
+``` bash
+composer require laizerox/php-wowemu-auth
+```
+
+## Usage
+
+### Register
+
+First you'll want to use Composer's `autoloader`. Place this at the top of your script.
+
+``` php
+require_once __DIR__ . '/vendor/autoload.php';
+use Laizerox\Wowemu\SRP\UserClient;
+```
+
+Next you'll need to create the verifier and salt values using the username and password which your user submitted on your registration form.
+
+``` php
+$client = new UserClient($username);
+$verifier = $client->generateVerifier($password);
+$salt = $client->generateSalt();
+```
+
+Once that is generated you'll just insert those values into the database to the `v` and `s` fields.
+
+### Login
+
+First you'll want to use Composer's `autoloader`. Place this at the top of your script.
+
+``` php
+require_once __DIR__ . '/vendor/autoload.php';
+use Laizerox\Wowemu\SRP\UserClient;
+```
+
+Next you'll need to generate your "verifier". Think of this as the hashed version of the password your user put into the password field of you login form.
+
+``` php
+$client = new UserClient($username, $saltFromDatabase);
+$verifier = strtoupper($client->generateVerifier($password));
+```
+
+Next you'll want to compare that value with the value stored in your CMaNGOS `realmd.account` table. You can see below for more of an example.
+
+## Examples
+
+### Register
+
+This example goes over how a user can register via a web form.
+
+``` php
+<?php
+
+/* register.php */
+
+require_once __DIR__ . '/vendor/autoload.php';
+use Laizerox\Wowemu\SRP\UserClient;
+
+/* Connect to your CMaNGOS database. */
+$db = new mysqli($dbHost, $dbUser, $dbPassword, $dbName);
+
+/* If the form has been submitted. */
+if (isset($_POST['register'])) {
+    $username = $_POST['username'];
+    $password = $_POST['password'];
+    
+    /* Grab the users IP address. */
+    $ip = $_SERVER['REMOTE_ADDR'];
+    
+    /* Set the join date. */
+    $joinDate = date('Y-m-d H:i:s');
+    
+    /* Set GM Level. */
+    $gmLevel = '0';
+
+    /* Set expansion pack - Wrath of the Lich King. */
+    $expansion = '2';
+    
+    /* Create your v and s values. */
+    $client = new UserClient($username);
+    $verifier = $client->generateVerifier($password);
```

Comment:
nice docs. thx for your effort :+1:  could you please move `$client->generateSalt()` before `$client->generateVerifier($password)`? otherwise `generateVerifier()` would raise an execption due to the fact that it's computation is based on salt value

Reply:
I don't believe that is correct, but checking now.

Reply:
Ok, moved it up so we should be all set now 👍🏼